ui/pages/asset/components/token-buttons.tsx

```diff
@@ -233,6 +240,13 @@ const TokenButtons = ({
           );
           try {
             await setCorrectChain();
+            if (token.chainId !== currentChainId) {
+              // TODO: Properly handle this case, for now we are stopping execution to prevent users from inadvertendly navigating to incompatible chains
```

Comment:
I understand that we agreed on doing this initially, so we make sure we do not proceed unless we have the right network;
But looking again into this; i think the intended behavior initially was if setCorrectChain fails it should into the throw skipping the `startNewDraftTransaction`;
This was not happening because:
1- we were not throwing inside the setCorrectChain fct 
2- the setActiveNetwork which is also not throwing if the switch network fails;

If we fix these two points, when the network switch fails, clicking on the button wont do anything (and we can work with Amanda to see how we want to display the error msg but it will prevent the user from proceeding in cases of network switch errors)